data-raw/create_sdtms_data.R

```diff
@@ -0,0 +1,129 @@
+
+# Helper function to retrieve the label attribute for a column
+#' @description Retrieve column label attribute or return a default.
+#' @param data The dataset containing the column.
+#' @param col_name The name of the column.
+#' @return A string containing the label attribute or "undocumented field".
+get_attr <- function(data, col_name) {
+  att <- attr(data[[col_name]], "label")
+  if (is.null(att) || att == "null") {
+    att <- "undocumented field"
+  }
+  return(att)
+}
+
+#' @description Create an HTML hyperlink for use in documentation, return plain text if the source is text,
+#' or a default message if the URL is invalid.
+#' @param url The URL to link to or plain text if the source is not a URL.
+#' @param link_text Optional text for the hyperlink. Defaults to an empty string.
+#' @return A string containing the HTML anchor tag, plain text if the source is text, or a default message for invalid URLs.
+generate_hyperlink <- function(url, link_text = "") {
+  # Define a basic regular expression for URL validation
+  url_pattern <- "^(https?://)"
+
+  # Check if the input is a valid URL
+  if (!nzchar(url) || is.null(url)) {
+    return("The source is inaccessible.")
+  } else if (!grepl(url_pattern, url)) {
+    # If the source is plain text (not a URL), return it as-is
+    return(url)
+  }
+
+  # Generate and return the hyperlink if the URL is valid
+  paste0('<a href="', url, '" target="_blank">', "Access the source of the ", link_text, " dataset.</a>")
+}
+
+
+# Main function to write documentation for a dataset
+#' @description Generate R documentation for a dataset.
+#' @param data The dataset object.
+#' @param dataset_name The name of the dataset.
+#' @param dataset_label The label for the dataset. Defaults to "No label available".
+#' @param dataset_description A description of the dataset. Defaults to "No description available".
+#' @param dataset_author The author of the dataset. Defaults to NULL if not available.
+#' @param dataset_source The source of the dataset, as a URL or text. Defaults to "No source available".
+write_doc <- function(data, dataset_name, dataset_label = "No label available",
+                      dataset_description = "No description available", dataset_author = NULL,
+                      dataset_source = "No source available") {
+  dataset_source <- generate_hyperlink(dataset_source, dataset_label)
+  doc_string <- paste(
+    "# This file is automatically generated by data-raw/create_sdtms_data.R.",
+    "# For updating it please edit inst/extdata/sdtms-specs.json and rerun create_sdtms_data.R.",
+    "# Manual edits are not recommended, as changes may be overwritten.",
+    sprintf("#' %s", dataset_label),
+    "#'",
+    sprintf("#' %s", dataset_description),
+    "#'",
+    sprintf("#' @name %s", dataset_name),
+    sprintf("#' @title %s", dataset_label),
+    sprintf("#' @keywords dataset"),
+    sprintf("#' @description dataset"),
```

Comment:
This looks strange:
![image](https://github.com/user-attachments/assets/7191e0d3-bdc7-451b-a69f-2ef1b582f465)